plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -0,0 +1,364 @@
+    });
+
+    const users = paginated<GitLabUser>(options => client.listUsers(options), {
+      page: 1,
+      per_page: 50,
+      active: true,
+    });
+
+    const groups = paginated<GitLabGroup>(
+      options => client.listGroups(options),
+      {
+        page: 1,
+        per_page: 50,
+      },
+    );
+
+    const idMappedGroup: { [groupId: number]: GitLabGroup } = {};
+
+    const res: Result = {
+      scanned: 0,
+      matches: [],
+    };
+
+    const groupRes: GroupResult = {
+      scanned: 0,
+      matches: [],
+    };
+
+    for await (const group of groups) {
+      if (!this.config.groupPattern.test(group.full_path ?? '')) {
+        continue;
+      }
+
+      groupRes.scanned++;
+      groupRes.matches.push(group);
+
+      idMappedGroup[group.id] = group;
+    }
+
+    for await (const user of users) {
+      if (!this.config.userPattern.test(user.email ?? '')) {
+        continue;
+      }
+
+      res.scanned++;
+
+      if (user.active) {
+        continue;
+      }
+
+      const memberships = await client.getUserMemberships(user.id);
+      const userGroups: GitLabGroup[] = [];
+
+      for (const i of memberships) {
+        if (
+          i.source_type === 'Namespace' &&
+          idMappedGroup.hasOwnProperty(i.source_id)
+        ) {
+          userGroups.push(idMappedGroup[i.source_id]);
+        }
+      }
+
+      user.groups = userGroups;
+
+      res.matches.push(user);
+    }
+
+    const groupsWithUsers = groupRes.matches.filter(group => {
+      return (
+        res.matches.filter(x => {
+          return !!x.groups?.find(y => y.id === group.id);
+        }).length > 0
+      );
+    });
+
+    const userEntities = res.matches.map(p =>
+      this.createUserEntity(p, this.integration.config.host),
+    );
+    const groupEntities = this.createGroupEntities(
+      groupsWithUsers,
+      this.integration.config.host,
+    );
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: [...userEntities, ...groupEntities].map(entity => ({
+        locationKey: this.getProviderName(),
+        entity: this.withLocations(this.integration.config.host, entity),
+      })),
+    });
+  }
+
+  private createGroupEntities(
+    groupResult: GitLabGroup[],
+    host: string,
+  ): GroupEntity[] {
+    const idMapped: { [groupId: number]: GitLabGroup } = {};
+    const entities: GroupEntity[] = [];
+
+    for (const group of groupResult) {
+      idMapped[group.id] = group;
+    }
+
+    for (const group of groupResult) {
+      const entity = this.createGroupEntity(group, host);
+
+      if (group.parent_id && idMapped.hasOwnProperty(group.parent_id)) {
+        entity.spec.parent = idMapped[group.parent_id].full_path;
+      }
+
+      entities.push(entity);
+    }
+
+    return entities;
+  }
+
+  private withLocations(host: string, entity: Entity): Entity {
+    const location =
+      entity.kind === 'Group'
+        ? `url:${host}/teams/${entity.metadata.name}`
```

Comment:
Maybe URL encode these?

Reply:
I've checked for the gitlab repository provider and the github provider, both don't encode that.